### PR TITLE
Import token after review step

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

To allow users to import a custom token, the canister IDs need to be entered, reviewed by the user, and then saved to the NNS-dapp canister. This update introduces the review step and ensures that the token is saved after user confirmation.

# Changes

- Display step 2 (review) when the “Next” button is clicked on step 1.
- Call `addImportToken` when the “Confirm” button is clicked on step 2.

# Tests

- Unit tests have been added.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts [hunk 1]
```diff
@@ -222,4 +232,185 @@ describe("ImportTokenModal", () => {
...
+      importedTokens: [
+        {
+          index_canister_id: [indexCanisterId],
+          ledger_canister_id: ledgerCanisterId,
+        },
+      ],
+    });
+    expect(getImportedTokensSpy).toBeCalledTimes(2);
+
+    expect(get(pageStore)).toEqual({
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts:355]: Should this be here or is it covered by the next test?
If it should be here, can you also add an expect before the navigation?

File: frontend/src/lib/modals/accounts/ImportTokenModal.svelte [hunk 2]
```diff
@@ -113,6 +121,40 @@
 
     next();
   };
+
+  const onConfirm = async () => {
+    if (
+      isNullish(ledgerCanisterId) ||
+      isNullish($importedTokensStore.importedTokens)
+    ) {
+      return;
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/modals/accounts/ImportTokenModal.svelte:131]: Correct. A comment added.

File: frontend/src/lib/stores/icrc-canisters.store.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { writable } from "svelte/store";
 
 export interface IcrcCanisters {
   ledgerCanisterId: Principal;
-  indexCanisterId: Principal;
+  indexCanisterId: Principal | undefined;
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/stores/icrc-canisters.store.ts]: Nope, that's from another. Removed.

File: frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts [hunk 4]
```diff
@@ -43,6 +47,7 @@ describe("ImportTokenModal", () => {
     return {
       po,
       formPo: po.getImportTokenFormPo(),
+      reviewPo: po.getImportTokenReviewPo(),
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts]: Agree, removed.

File: frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts [hunk 5]
```diff
@@ -222,4 +232,185 @@ describe("ImportTokenModal", () => {
...
+    expect(await reviewPo.getLedgerCanisterIdPo().getCanisterIdText()).toEqual(
+      ledgerCanisterId.toText()
+    );
+    expect(await reviewPo.getIndexCanisterIdPo().getCanisterIdText()).toEqual(
+      indexCanisterId.toText()
+    );
+    expect(await reviewPo.getTokenName()).toEqual(tokenMetaData.name);
+    expect(await reviewPo.getTokenSymbol()).toEqual(tokenMetaData.symbol);
+    expect(await reviewPo.getLogoSource()).toEqual(tokenMetaData.logo);
+  });
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts:260]: Added.

File: frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts [hunk 6]
```diff
@@ -222,4 +232,185 @@ describe("ImportTokenModal", () => {
...
+    expect(setImportedTokensSpy).toBeCalledWith({
+      identity: mockIdentity,
+      importedTokens: [
+        {
+          index_canister_id: [indexCanisterId],
+          ledger_canister_id: ledgerCanisterId,
+        },
+      ],
+    });
+    expect(getImportedTokensSpy).toBeCalledTimes(2);
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts:362]: Added.

File: frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/accounts/ImportTokenModal.spec.ts]: There was one at the beginning of the test. But I agree that having it here is a redundant due to the next test. So the expectation has been removed.

**Review by dskloetd — APPROVED** [review]

**Pull request merged**